netplugin/plugin/netplugin.go

```diff
@@ -303,27 +304,57 @@ func (p *NetPlugin) Reinit(cfg Config) {
 }
 
 //InitGlobalSettings initializes cluster-wide settings (e.g. fwd-mode)
-func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) {
+func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) error {
+
+	/*
+	Query global settings from state store
+	1. if forward mode or private net is empty, retry 1 sec
+	2. if forward mode doesn't match, return error
+	3. if stored private net is wrong, return error
+	*/
 
 	gCfg := mastercfg.GlobConfig{}
 	gCfg.StateDriver = stateDriver
-	err := gCfg.Read("")
-	if err != nil {
-		logrus.Errorf("Error reading forwarding mode from cluster store")
-		inst.FwdMode = "bridge"
-		inst.HostPvtNW = defaultPvtSubnet
-		return
+
+	// wait until able to get fwd mode and private subnet
+	for {
+		err := gCfg.Read("")
+		if err != nil {
+			logrus.Errorf("Error reading global settings from cluster store")
+		}
+		if gCfg.FwdMode == "" || gCfg.PvtSubnet == "" {
+			if gCfg.FwdMode == "" {
+				logrus.Errorf("Error reading forwarding mode from cluster store")
+			} else {
+				logrus.Errorf("Error reading private subnet cluster store")
+			}
```

Comment:
Shouldn't it be:
`if gCfg.FwdMode == "" { logrus.Warnf }
  if gCfg.PvtSubnet == "" { logrus.Warnf} `